packages/react-server/src/ReactFlightServer.js

```diff
@@ -826,6 +833,7 @@ function performWork(request: Request): void {
   const prevCache = getCurrentCache();
   ReactCurrentDispatcher.current = Dispatcher;
   setCurrentCache(request.cache);
```

Comment:
Instead of setting up two different contextual variables (and more to come) we should just use the request for the Cache too. That can be a follow up but should fast follow.

While we're add it we should also put the Cache behind the `enableCache` since it's not coupled to Flight.